examples/example-kotlin/src/main/kotlin/io/grpc/examples/helloworld/HelloWorldClient.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2015, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.helloworld
+
+import io.grpc.ManagedChannel
+import io.grpc.ManagedChannelBuilder
+import io.grpc.StatusRuntimeException
+import java.util.concurrent.TimeUnit
+import java.util.logging.Level
+import java.util.logging.Logger
+
+/**
+ * A simple client that requests a greeting from the [HelloWorldServer].
+ */
+class HelloWorldClient
+/** Construct client for accessing RouteGuide server using the existing channel.  */
+internal constructor(private val channel: ManagedChannel) {
+    private val blockingStub: GreeterGrpc.GreeterBlockingStub
+
+    /** Construct client connecting to HelloWorld server at `host:port`.  */
+    constructor(host: String, port: Int) : this(ManagedChannelBuilder.forAddress(host, port)
+            // Channels are secure by default (via SSL/TLS). For the example we disable TLS to avoid
+            // needing certificates.
+            .usePlaintext(true)
+            .build()) {
+    }
+
+    init {
+        blockingStub = GreeterGrpc.newBlockingStub(channel)
+    }
+
+    @Throws(InterruptedException::class)
+    fun shutdown() {
+        channel.shutdown().awaitTermination(5, TimeUnit.SECONDS)
+    }
+
+    /** Say hello to server.  */
+    fun greet(name: String) {
+        logger.info("Will try to greet $name ...")
+        val request = HelloRequest.newBuilder().setName(name).build()
+        val response: HelloReply
+        try {
+            response = blockingStub.sayHello(request)
+        } catch (e: StatusRuntimeException) {
+            logger.log(Level.WARNING, "RPC failed: {0}", e.status)
+            return
+        }
+
+        logger.info("Greeting: " + response.message)
```

Comment:
It seems WARNING has higher level than INFO
```kt
logger.warning("RPC failed: ${e.status}")
```
and
```kt
// avoid stringcat
logger.log(Level.INFO, "Greeting: {0}", response.message)
```

Reply:
Isn't that normal for errors to be higher level than info?

Reply:
My point is to avoid string concatenation when/if the log is (more likely) not to show up.

`logger.info("Greeting: " + response.message)` or `logger.info("Greeting:  ${response.message}")`  is always running string concatenation first even if the log is not going to show up if I set the log level higher than INFO.